resources/js/components/JournalListComponent.vue

```diff
@@ -30,188 +30,255 @@
+          });
+      },
+      showDetailsModal(id) {
+        this.showJournalDetailsModal = true;
+        this.editJournalId = id;
+        axios.get("api/journals/" + id)
+          .then(response => {
+            this.editJournal = response.data;
+          })
+          .catch(error => {
+            if (error.response.status) {
+              alert("Server Error");
+            }
+          });
+      },
+      showDeleteModal(id) {
+        this.showConfirmationModal = true;
+        this.deleteJournalId = id;
+      },
+      createJournal(date, text, plan_tomorrow, goal_tomorrow, goal_status) {
+        axios
+          .post("api/journals", {
+            user_id: this.userid,
+            date: moment(date).format("YYYY-MM-DD"),
+            text: text,
+            plan_tomorrow: plan_tomorrow,
+            goal_tomorrow: goal_tomorrow,
+            goal_status: goal_status
+          })
+          .then(({data}) => {
+            this.journals.push(new Journal(data));
+            this.showJournalCreateModal = false;
+          })
+          .catch(error => {
+            if (error.response.status == 422) {
+              let msgs = this.messages;
+              let errors = error.response.data.errors;
+
+              Object.keys(errors).forEach(key => {
+                errors[key].forEach(function (error) {
+                  msgs.push(error);
+                });
+              });
+              this.messages = msgs;
+            } else {
+              this.messages = ["Server error."];
+            }
+          });
+      },
+      deleteJournal(id) {
+        axios
+          .delete("api/journals/" + id)
+          .then(response => {
+            let index = this.journals.findIndex(journal => journal.id === id);
+            this.journals.splice(index, 1);
+            this.showConfirmationModal = false;
+          })
+          .catch(error => {
+            if (error.response.status) {
+              alert("Server Error");
+            }
+          });
+      },
+      edit(editJournalId,date, text, plan_tomorrow, goal_tomorrow, goal_status) {
+        console.log(editJournalId,date,text,plan_tomorrow,goal_tomorrow,goal_status)
+        axios.put("api/journals/" + editJournalId, {
           user_id: this.userid,
           date: moment(date).format("YYYY-MM-DD"),
           text: text,
           plan_tomorrow: plan_tomorrow,
           goal_tomorrow: goal_tomorrow,
           goal_status: goal_status
         })
-        .then(({ data }) => {
-          this.journals.push(new Journal(data));
-          this.showJournalCreateModal = false;
-        })
-        .catch(error => {
-          if (error.response.status == 422) {
-            let msgs = this.messages;
-            let errors = error.response.data.errors;
-
-            Object.keys(errors).forEach(key => {
-              errors[key].forEach(function(error) {
-                msgs.push(error);
+          .then(response => {
+            alert("Success");
```

Comment:
Please remove `alert("Success")`. We want to update the journals list. 
Find the index of the edited journal and updates its values.